rotkehlchen/api/rest.py

```diff
@@ -3465,7 +3465,24 @@ def get_historical_assets_price(
             self,
             assets_timestamp: list[tuple[Asset, Timestamp]],
             target_asset: Asset,
+            only_cache_period: int | None = None,
     ) -> dict[str, Any]:
+        if only_cache_period is not None:
+            result: dict[str, Any] = {
+                'assets': defaultdict(lambda: defaultdict(lambda: ZERO_PRICE)),
+                'target_asset': target_asset.identifier,
+            }
+            for price_result, query in zip(GlobalDBHandler.get_historical_prices(
+                query_data=[(entry[0], target_asset, entry[1]) for entry in assets_timestamp],
+                max_seconds_distance=only_cache_period,
+            ), assets_timestamp, strict=False):
+                if price_result is None:
+                    result['assets'][query[0]][query[1]] = None
+                else:
```

Comment:
As mentioned in discord. Just omit timestamps for which price is not found.

```suggestion
                if price_result is not None:
```